data/data/bootstrap/baremetal/files/etc/containers/systemd/image-customization.container

```diff
@@ -5,6 +5,7 @@ Requires=build-ironic-env.service extract-machine-os.service
 PartOf=ironic.service
 Wants=network-online.target
 After=network-online.target ironic-volume.service build-ironic-env.service extract-machine-os.service
+OnSuccess=ironic.service
```

Comment:
The man page for systemd says that the `OnSuccess` units are started 'when this unit enters the "inactive" state.' Since this is a simple service with `Restart=always`, I wouldn't expect that to ever happen.

Theroetically since this is not a oneshot service it shouldn't be subject to the same problem and this line wouldn't be required (if I understand the problem correctly).

Reply:
From what I observed without this line (but with OnSuccess=image-customization.service in extract-machine-os), when "extract-machine-os" finished (after an initial failure) ,
it triggers image-customization to run but it can then fail to start because the container image takes too long to download, when it finally does complete the only way I could get it to trigger ironic.service is with this line.

Reply:
In addition do doing what I wanted looks like this also has some unwanted effects, causing ironic to be restarted after its finished

```
Fri 2024-08-30 09:40:49 UTC localhost.localdomain init.scope[1]: image-customization.service: Deactivated successfully.
Fri 2024-08-30 09:40:49 UTC localhost.localdomain init.scope[1]: Stopped Customized Machine OS Image Server.
Fri 2024-08-30 09:40:49 UTC localhost.localdomain init.scope[1]: image-customization.service: Consumed 6.895s CPU time.
Fri 2024-08-30 09:40:49 UTC localhost.localdomain init.scope[1]: image-customization.service: Triggering OnSuccess= dependencies.
Fri 2024-08-30 09:40:49 UTC localhost.localdomain init.scope[1]: Starting Customized Machine OS Image Server...
Fri 2024-08-30 09:40:49 UTC localhost.localdomain init.scope[1]: Starting DHCP Service for Provisioning Network...
Fri 2024-08-30 09:40:49 UTC localhost.localdomain init.scope[1]: Starting HTTP Server for Ironic...
Fri 2024-08-30 09:40:49 UTC localhost.localdomain image-customization.service[281984]: a8f6a07a0753b4b762c955cb4
```

I'm starting to thin we should bump up the timeouts for both  extract-machine-os and image-customization to deal with the situation without changing any other behaviour.